packages/sanity/src/structure/panes/document/timeline/expandableTimelineItemMenu.tsx

```diff
@@ -0,0 +1,88 @@
+import {Menu, usePortal} from '@sanity/ui'
+import {type MouseEvent, useCallback} from 'react'
+import {ContextMenuButton, useTranslation} from 'sanity'
+
+import {MenuButton, MenuItem} from '../../../../ui-components'
+import {structureLocaleNamespace} from '../../../i18n'
+import {TIMELINE_LIST_WRAPPER_ID} from './timeline'
+import {TIMELINE_MENU_PORTAL} from './timelineMenu'
+
+/**
+ * This is a hack to force the scrollbar to not appear when the list is expanding,
+ * if we don't do this the scrollbar will appear for a brief moment when the list is expanding and then disappear
+ * when the list is fully expanded.
+ */
+function hideScrollbarOnExpand(isExpanded: boolean) {
```

Comment:
Explanation on why this is needed, thoughts and suggestions are welcomed! 
https://www.loom.com/share/046f783e048f463e84e6e6fd31e551b7

Reply:
This is a really ingenious solution but I have to admit that I hate it though I understand the need 😭 
While I can potentially think of more elegant ways of fixing this (we do have sections of the code that uses the ResizeObserver) in the end I think we would still need to end up changing the styles like you're doing here.

If anyone else thinks of a better solution (Herman might actually!) I'd also like to know 👀

Reply:
Will give it a try to the resizeObserver! I was not aware of that, thanks!

Reply:
Hey @RitaDias, thanks for the suggestion, I tried it but it didn't work in the way I need it, I think it's because the `onResize` callback is triggered by an useEffect and that useEffect is running after the element has changed it's scroll size.
So the bar is still visible when expanding.

I think we will have to go with this not nice idea but it does the job. Unless we can think of something else 😄 .

Reply:
Yeah I'd assume that would be the case!
I'm approving it, the way I see it I think we can go ahead with this solution and rework it at a later point if we feel it's not good enough to go to production we'll handle it then :)